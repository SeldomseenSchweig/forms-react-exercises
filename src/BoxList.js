import React, {useState} from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';
import { v4 as uuid } from 'uuid'




const BoxList =()=>{
    const INITIAL_STATE = [
        { 
            id:uuid(),
            color: 'red',
            height:10,
            width:10
        },
        { 
            id:uuid(),
            color: 'blue',
            height:10,
            width:10
        }
    ]
    const [boxes, setBoxes] = useState(INITIAL_STATE);
    const addBox = (newBox)=>{
        setBoxes(boxes =>([...boxes,{...newBox, id:uuid()} ]));
    }
    const remove = (e) => {
        
        const id = e.target.parentNode.id;
                    
             setBoxes((boxes)=>{ 
                const boxesCopy = [...boxes]
                const indexOfObject = boxesCopy.findIndex( object => object.id === id )
                const final = boxesCopy.splice(indexOfObject, 1);
                return [...boxesCopy]
                



             } )
        }
        
    



    return (

        <div>
            <NewBoxForm addBox={addBox} />
            <h1> Box List</h1>
            <div>
            
            { boxes.map(({id,color,height,width}) =><Box id={id} color={color} height={height} width={width} key={id} remove={remove}/> )}
            </div>

            
        </div>
    )



}

export default BoxList;