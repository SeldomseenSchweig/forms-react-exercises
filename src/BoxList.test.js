import React from "react";
import { render, fireEvent, queryByTestId, queryAllByTestId } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import BoxList from "./BoxList";


test("renders without crashing", function (){

    render(<BoxList />);
});

test("matches snapshot", function () {
    const {asFragment} = render(<BoxList />)
    expect (asFragment()).toMatchSnapshot();
});

test("should add new item", ()=>{

    const {queryByText, getByLabelText, queryByTestId} = render(<BoxList />)
    const input = getByLabelText('Color')
    const button = queryByText('Submit')
    expect(queryByTestId('purple')).not.toBeInTheDocument();
    fireEvent.change(input,{target:{value: "purple"}});
    fireEvent.click(button);
    expect(queryByTestId('purple')).toBeInTheDocument();

})


