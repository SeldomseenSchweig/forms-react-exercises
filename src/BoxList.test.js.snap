// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="color"
      >
         Color
      </label>
      <input
        id="color"
        name="color"
        placeholder="Color"
        type="text"
        value=""
      />
      <label
        for="height"
      >
         Box height
      </label>
      <input
        id="height"
        name="height"
        type="number"
        value=""
      />
      <label
        for="width"
      >
         width 
      </label>
      <input
        id="width"
        name="width"
        type="number"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
    <h1>
       Box List
    </h1>
    <div>
      <div
        class="Box"
        data-testid="red"
        id="2"
        style="background-color: red; min-height: 3em; min-width: 3em; max-height: 10em; max-width: 10em; width: 10em; height: 10em;"
      >
        <button>
          X
        </button>
      </div>
      <div
        class="Box"
        data-testid="blue"
        id="1"
        style="background-color: blue; min-height: 3em; min-width: 3em; max-height: 10em; max-width: 10em; width: 10em; height: 10em;"
      >
        <button>
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
